### PR TITLE
Add ipconfigPORT_SUPPRESS_WARNING to suppress warning messages in portable layer.

### PR DESCRIPTION
**Pull request by ActoryOu**
Fix issue that #967 mentioned.
1. Remove redundant warning messages to avoid always warning in any platform.
2. Add ipconfigPORT_SUPPRESS_WARNING to suppress warning messages in portable layer.

Description
-----------

- Add ipconfigPORT_SUPPRESS_WARNING to suppress warning messages in portable layer.
- Remove redundant warning messages.
- Replace some warning messages with comments.

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c [hunk 1]
```diff
@@ -662,7 +665,10 @@ XStatus init_dma( xemacpsif_s * xemacpsif )
         #if ( ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM != 0 )
             value |= XEMACPS_DMACR_TCPCKSUM_MASK;
         #else
-        #warning Are you sure the EMAC should not calculate outgoing checksums?
+            #if ( ipconfigPORT_SUPPRESS_WARNING != 0 )
+            #warning Are you sure the EMAC should not calculate outgoing checksums?
```
Comment (tony-josi-aws) [inline source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c:669]: Indentation

File: source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c [hunk 2]
```diff
@@ -678,7 +684,10 @@ XStatus init_dma( xemacpsif_s * xemacpsif )
         #if ( ipconfigDRIVER_INCLUDED_RX_IP_CHECKSUM != 0 )
             value |= XEMACPS_NWCFG_RXCHKSUMEN_MASK;
         #else
-        #warning Are you sure the EMAC should not calculate incoming checksums?
+            #if ( ipconfigPORT_SUPPRESS_WARNING != 0 )
+            #warning Are you sure the EMAC should not calculate incoming checksums?
```
Comment (tony-josi-aws) [inline source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c:688]: Indentation

File: source/portable/NetworkInterface/Zynq/NetworkInterface.c [hunk 3]
```diff
@@ -327,7 +331,7 @@ static BaseType_t xZynqNetworkInterfaceInitialise( NetworkInterface_t * pxInterf
             uint32_t ulValue = XEmacPs_ReadReg( pxEMAC_PS->Config.BaseAddress, XEMACPS_NWCFG_OFFSET );
             /* Allow the use of hashed MAC addresses. */
             ulValue |= XEMACPS_NWCFG_MCASTHASHEN_MASK;
-            #warning As 'MCASTHASHEN' doesn't seem to work, use the promiscuous mode so that IPv6 multicast packets are received.
+            /* As 'MCASTHASHEN' doesn't seem to work, use the promiscuous mode so that IPv6 multicast packets are received. */
```
Comment (tony-josi-aws) [inline source/portable/NetworkInterface/Zynq/NetworkInterface.c:334]: Can we keep this warning inside the new `ipconfigPORT_SUPPRESS_WARNING` config check? And remove the the apostrophe in word `doesn't`.

File: source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c [hunk 4]
```diff
@@ -59,15 +59,21 @@
 
 /* Check for optimal performance parameters */
 #if ( CONF_GMAC_NCFGR_RXCOEN == 0 )
-    #warning This driver works best with RX CRC offloading enabled.
+    #if ( ipconfigPORT_SUPPRESS_WARNING != 0 )
```
Comment (tony-josi-aws) [inline source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c:62]: Shouldn't the checks be `#if ( ipconfigPORT_SUPPRESS_WARNING == 0 )`

Comment (ActoryOu) [reply source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c:62]: This configuration is used to "suppress" the warning messages. So we should skip warning users when it's 1.

File: source/portable/NetworkInterface/Zynq/NetworkInterface.c (hunk 3 above)
Comment (ActoryOu) [reply source/portable/NetworkInterface/Zynq/NetworkInterface.c:334]: I don't think it's valid to keep this warning message always printing. Any warning message should be handled by configurations. And I don't think it's valid to warn users about this.

File: source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c (hunk 1 above)
Comment (ActoryOu) [reply source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c]: Updated. It's modified by uncrustify. I added '{}' to make indentation right.

File: source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c (hunk 2 above)
Comment (ActoryOu) [reply source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_dma.c]: Updated. It's modified by uncrustify. I added `{}` to make indentation right.

File: source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c (hunk 4 above)
Comment (tony-josi-aws) [reply source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c:62]: Yes, but `#if ( ipconfigPORT_SUPPRESS_WARNING != 0 )` won't skip the warning when `ipconfigPORT_SUPPRESS_WARNING` is set to 1.

Comment (ActoryOu) [reply source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c:62]: Oops. My fault. Let me update it once again.

**Review by tony-josi-aws — DISMISSED** [review]

File: source/portable/NetworkInterface/ksz8851snl/NetworkInterface.c [hunk 5]
```diff
@@ -939,11 +941,11 @@ static void ksz8851snl_update()
                int rxHead = xMicrelDevice.us_rx_head;
                /* RX step18-19: pad with dummy data to keep dword alignment. */
                /* Packet lengths will be rounded up to a multiple of "sizeof size_t". */
-/*			xLength = xMicrelDevice.rx_buffers[ rxHead ]->xDataLength & 3; */
-/*			if( xLength != 0 ) */
-/*			{ */
-/*				ksz8851_fifo_dummy( 4 - xLength ); */
-/*			} */
+               /* xLength = xMicrelDevice.rx_buffers[ rxHead ]->xDataLength & 3; */
```
Comment (moninom1) [inline source/portable/NetworkInterface/ksz8851snl/NetworkInterface.c:944]: Why do we have /* */ in every line, it should be one covering all as mentioned in style guideline https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html

Comment (ActoryOu) [reply source/portable/NetworkInterface/ksz8851snl/NetworkInterface.c]: Updated

**Review by tony-josi-aws — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**